modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -142,7 +148,8 @@ public void onFailure(Exception t) {
 
             @Override
             protected void doRun() throws Exception {
-                client.performRequestAsync(method, uri, params, entity, new ResponseListener() {
+                HeapBufferedAsyncResponseConsumer consumer = new HeapBufferedAsyncResponseConsumer(BUFFER_LIMIT);
```

Comment:
I also think it'd be good to have `HeapBufferedAsyncResponseConsumer` take a `ByteSizeValue` instead of an `int` of bytes, much less chance of misinterpretation.